### PR TITLE
MGMT-13001: Adding Equinix infrastructure

### PR DESCRIPTION
**Pull request by danmanor**
Adding the basic infrastructure for Equinix related metrics + basic Equinix metrics

### CODE REVIEW

File: src/jobsautoreport/models.py [hunk 1]
```diff
@@ -8,6 +8,6 @@ class JobType(Enum):
     BATCH = "batch"
 
 
-class JobState(Enum):
+class ObjectState(Enum):
```
Comment (adriengentil) [inline src/jobsautoreport/models.py:11]:
```suggestion
class JobState(Enum):
...

StepState = NewType("StepState", JobState)
```
The name `ObjectState` is too vague to me. You can create an alias with `NewType` https://docs.python.org/3/library/typing.html#newtype.

Or you can just name it like `ProwExecutionState` maybe?

File: src/jobsautoreport/query.py [hunk 2]
```diff
@@ -35,21 +37,30 @@ def _get_query_all_jobs(from_date: datetime, to_date: datetime) -> dict:
...
+        objects = self._scan(query, self._jobs_index)
+        return utils.parse_jobs(objects)
 
-    def _query_and_log(self, query: dict) -> list[JobDetails]:
+    def _query_step_events_and_log(self, query: dict) -> list[StepEvent]:
         logger.debug("OpenSearch query: %s", query)
-        jobs = self._scan(query)
-        return utils.parse_jobs(jobs)
+        objects = self._scan(query, self._steps_index)
+        return utils.parse_step_events(objects)
```
Comment (adriengentil) [inline src/jobsautoreport/query.py:58]: maybe we can filter here directly on the steps matching `packet-setup`, as there are much more steps than jobs, and we are really interested into one part of the data

File: src/jobsautoreport/report.py [hunk 3]
```diff
@@ -114,7 +120,13 @@ def _is_e2e_or_subsystem_class(job: JobDetails) -> bool:
...
 
     def get_report(self, from_date: datetime, to_date: datetime) -> Report:
-        jobs = self._querier.query_all_jobs(from_date, to_date)
+        jobs = self._querier.query_jobs(from_date, to_date)
+        step_events = self._querier.query_step_events(from_date, to_date)
+        assisted_step_events = [
+            step_event
+            for step_event in step_events
+            if self._is_assisted_repository(step_event.job)
+        ]
```
Comment (adriengentil) [inline src/jobsautoreport/report.py:129]: don't filter on assisted repo, we want all repos that may lease machines from equinix (... and we pay for them).

File: src/jobsautoreport/utils.py [hunk 4]
```diff
@@ -1,12 +1,19 @@
 from typing import Any
```
Comment (adriengentil) [inline src/jobsautoreport/utils.py:1]: I think all the functions in this file can end up in the querier. They don't look to be used anywhere else?

**Review by adriengentil — COMMENTED** [review]
few comments, looks good overall!

File: src/jobsautoreport/query.py [hunk 5]
```diff
@@ -35,21 +36,73 @@ def _get_query_all_jobs(from_date: datetime, to_date: datetime) -> dict:
...
         res = helpers.scan(
             client=self._os_client,
             query=query,
-            index=self._index,
+            index=index_name,
         )
         return list(res)
+
+    @classmethod
+    def parse_jobs(cls, data: list[dict[Any, Any]]) -> list[JobDetails]:
```
Comment (adriengentil) [inline src/jobsautoreport/query.py:95]:
```suggestion
    def _parse_jobs(cls, data: list[dict[Any, Any]]) -> list[JobDetails]:
```

File: src/jobsautoreport/query.py [hunk 6]
```diff
@@ -35,21 +36,73 @@ def _get_query_all_jobs(from_date: datetime, to_date: datetime) -> dict:
...
+        return [cls.parse_job(job["_source"]["job"]) for job in data]
+
+    @classmethod
+    def parse_step_events(cls, data: list[dict[Any, Any]]) -> list[StepEvent]:
```
Comment (adriengentil) [inline src/jobsautoreport/query.py:99]:
```suggestion
    def _parse_step_events(cls, data: list[dict[Any, Any]]) -> list[StepEvent]:
```

File: src/jobsautoreport/query.py [hunk 7]
```diff
@@ -35,21 +36,73 @@ def _get_query_all_jobs(from_date: datetime, to_date: datetime) -> dict:
...
+        return [cls.parse_step_event(step_event["_source"]) for step_event in data]
+
+    @staticmethod
+    def parse_job(data: dict) -> JobDetails:
```
Comment (adriengentil) [inline src/jobsautoreport/query.py:103]:
```suggestion
    def _parse_job(data: dict[Any, Any]) -> JobDetails:
```

File: src/jobsautoreport/query.py [hunk 8]
```diff
@@ -35,21 +36,73 @@ def _get_query_all_jobs(from_date: datetime, to_date: datetime) -> dict:
...
+        return JobDetails.parse_obj(data)
+
+    @staticmethod
+    def parse_step_event(data: dict) -> StepEvent:
```
Comment (adriengentil) [inline src/jobsautoreport/query.py:107]:
```suggestion
    def _parse_step_event(data: dict[Any, Any]) -> StepEvent:
```

File: src/jobsautoreport/query.py [hunk 9]
```diff
@@ -35,21 +36,73 @@ def _get_query_all_jobs(from_date: datetime, to_date: datetime) -> dict:
             }
         }
 
-    def query_all_jobs(
-        self, from_date: datetime, to_date: datetime
-    ) -> list[JobDetails]:
+    @staticmethod
+    def _get_query_all_steps(from_date: datetime, to_date: datetime) -> dict:
```
Comment (adriengentil) [inline src/jobsautoreport/query.py:40]:
```suggestion
    def _get_query_steps_by_name(from_date: datetime, to_date: datetime, name: str) -> dict[str, Any]:
```
could be good to add a parameter for the step name

File: src/jobsautoreport/query.py [hunk 10]
```diff
@@ -35,21 +36,73 @@ def _get_query_all_jobs(from_date: datetime, to_date: datetime) -> dict:
...
+                }
+            }
+        }
+
+    def query_jobs(self, from_date: datetime, to_date: datetime) -> list[JobDetails]:
         query = self._get_query_all_jobs(from_date, to_date)
-        return self._query_and_log(query)
+        return self._query_jobs_and_log(query)
+
+    def query_step_events(
```
Comment (adriengentil) [inline src/jobsautoreport/query.py:70]:
```suggestion
    def query_packet_setup_step_events(
```

File: src/jobsautoreport/query.py [hunk 11]
```diff
@@ -35,21 +36,73 @@ def _get_query_all_jobs(from_date: datetime, to_date: datetime) -> dict:
...
         logger.debug("OpenSearch query: %s", query)
-        jobs = self._scan(query)
-        return utils.parse_jobs(jobs)
+        objects = self._scan(query, self._jobs_index)
+        return self.parse_jobs(objects)
 
-    def _scan(self, query: dict[str, Any]) -> list[dict[Any, Any]]:
+    def _query_step_events_and_log(self, query: dict) -> list[StepEvent]:
+        logger.debug("OpenSearch query: %s", query)
+        objects = self._scan(query, self._steps_index)
```
Comment (adriengentil) [inline src/jobsautoreport/query.py:83]:
```suggestion
        es_steps = self._scan(query, self._steps_index)
```
try to not use much 'data', 'object' and all other too generic names

File: src/jobsautoreport/query.py [hunk 12]
```diff
@@ -35,21 +36,73 @@ def _get_query_all_jobs(from_date: datetime, to_date: datetime) -> dict:
...
+    ) -> list[StepEvent]:
+        query = self._get_query_all_steps(from_date, to_date)
+        return self._query_step_events_and_log(query)
 
-    def _query_and_log(self, query: dict) -> list[JobDetails]:
+    def _query_jobs_and_log(self, query: dict) -> list[JobDetails]:
         logger.debug("OpenSearch query: %s", query)
-        jobs = self._scan(query)
-        return utils.parse_jobs(jobs)
+        objects = self._scan(query, self._jobs_index)
```
Comment (adriengentil) [inline src/jobsautoreport/query.py:78]:
```suggestion
        es_steps = self._scan(query, self._jobs_index)
```

File: src/jobsautoreport/query.py [hunk 13]
```diff
@@ -35,21 +36,73 @@ def _get_query_all_jobs(from_date: datetime, to_date: datetime) -> dict:
...
+        return self._query_jobs_and_log(query)
+
+    def query_step_events(
+        self, from_date: datetime, to_date: datetime
+    ) -> list[StepEvent]:
+        query = self._get_query_all_steps(from_date, to_date)
+        return self._query_step_events_and_log(query)
 
-    def _query_and_log(self, query: dict) -> list[JobDetails]:
+    def _query_jobs_and_log(self, query: dict) -> list[JobDetails]:
```
Comment (adriengentil) [inline src/jobsautoreport/query.py:76]:
```suggestion
    def _query_jobs_and_log(self, query: dict[str, Any]) -> list[JobDetails]:
```

File: src/jobsautoreport/query.py [hunk 14]
```diff
@@ -35,21 +36,73 @@ def _get_query_all_jobs(from_date: datetime, to_date: datetime) -> dict:
...
         logger.debug("OpenSearch query: %s", query)
-        jobs = self._scan(query)
-        return utils.parse_jobs(jobs)
+        objects = self._scan(query, self._jobs_index)
+        return self.parse_jobs(objects)
 
-    def _scan(self, query: dict[str, Any]) -> list[dict[Any, Any]]:
+    def _query_step_events_and_log(self, query: dict) -> list[StepEvent]:
```
Comment (adriengentil) [inline src/jobsautoreport/query.py:81]:
```suggestion
    def _query_step_events_and_log(self, query: dict[str, Any]) -> list[StepEvent]:
```

**Comment by osherdp** [comment]
@danmanor you no longer need to do ``ok-to-test`` in this repo (unless someone removes you from the ``openshift-assisted`` org)

**Pull request merged**